### PR TITLE
common: Make adding new boards more generic

### PR DESCRIPTION
**Pull request by mkasenberg**
Move reset commands of boards to separate module files.
Replace nrfjprog in searching for a free device serial port and jlink serial number.
Add build and flash for dialog_da1469x-dk-pro.
Add tty_file and debugger_srn options to config of bot.

### CODE REVIEW

**Comment by mkasenberg** [comment]
@jrotkiewicz

File: autoptsclient_common.py [hunk 1]
```diff
@@ -1260,19 +1264,11 @@ def check_args(self, args):
...
-                sys.exit("%s is not a TTY nor COM file!" % repr(tty_file))
-            elif not os.path.exists(tty_file) and \
-                    (tty_file.startswith('/dev/ttyS') and
-                     not os.path.exists('COM' + str(int(tty_file["/dev/ttyS".__len__():]) + 1))):
-                sys.exit("%s TTY file does not exist!" % repr(tty_file))
+            if not tty_exists(args.tty_file):
+                sys.exit("%s serial port does not exist!" % repr(args.tty_file))
+
+            if args.tty_file.startswith("COM"):
+                args.tty_file = com_to_tty(args.tty_file)
```
Comment (michal-narajowski) [inline autoptsclient_common.py:1271]: We should handle error here

File: bot/common.py [hunk 2]
```diff
@@ -136,7 +137,7 @@ def run_tests(self, args, iut_config):
 
             _args[config].test_cases = value.get('test_cases', [])
 
-            if 'overlay' in value:
+            if 'overlay' in value and config != config_default:
```
Comment (michal-narajowski) [inline bot/common.py:140]: Is this change related?

File: ptsprojects/boards/__init__.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        """Reset HW DUT board with
+        """
+        log("About to reset DUT: %r", self.reset_cmd)
+
+        reset_process = subprocess.Popen(shlex.split(self.reset_cmd),
+                                         shell=False,
+                                         stdout=self.iutctl.iut_log_file,
+                                         stderr=self.iutctl.iut_log_file)
+
+        if reset_process.wait():
```
Comment (michal-narajowski) [inline ptsprojects/boards/__init__.py:53]: Needs a timeout argument.

File: ptsprojects/testcase_db.py [hunk 4]
```diff
@@ -7,7 +7,7 @@ class TestCaseTable:
     def __init__(self, name, database_file=DATABASE_FILE):
         self.database_file = database_file
         self._open()
-        self.name = name
+        self.name = name.replace(r'-', r'_')
```
Comment (michal-narajowski) [inline ptsprojects/testcase_db.py:10]: It seems that this is not related. If this solves some issue please create a separate commit and describe the problem.

File: pybtp/iutctl_common.py [hunk 5]
```diff
@@ -16,6 +16,7 @@
 import logging
 import os
 import queue
+import re
```
Comment (michal-narajowski) [inline pybtp/iutctl_common.py:19]: Unnecessary change.

File: pybtp/iutctl_common.py [hunk 6]
```diff
@@ -149,6 +150,7 @@ def __init__(self):
         self.event_handler_cb = None
 
     def _rx_task(self):
+
```
Comment (michal-narajowski) [inline pybtp/iutctl_common.py:153]: Unnecessary change.

File: bot/common.py (hunk 2 above)
Comment (mkasenberg) [reply bot/common.py]: Old workaround

**Review by michal-narajowski — APPROVED** [review]
